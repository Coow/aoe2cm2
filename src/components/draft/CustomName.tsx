import * as React from "react";

interface IProps {
    name: string
}

class CustomName extends React.Component<IProps, object> {
    public render() {
        const trimmedName = this.props.name.trim();
        if (trimmedName.length > 32) {
            return <abbr title={trimmedName}>{trimmedName.substring(0, 30) + 'â€¦'}</abbr>
        }
        return <span>{trimmedName}</span>
    }
}

export default CustomName;
