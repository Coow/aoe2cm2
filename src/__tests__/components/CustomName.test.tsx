import * as enzyme from 'enzyme';
import * as React from 'react';
import CustomName from "../../components/draft/CustomName";

it('Short name is not shortened', () => {
    const customName = enzyme.shallow(<CustomName name={'Player Name'}/>);
    expect(customName).toMatchSnapshot();
});

it('Long name is shortened', () => {
    const customName = enzyme.shallow(<CustomName name={'123456789012345678901234567890123456789'}/>);
    expect(customName).toMatchSnapshot();
});

it('32 long name is not shortened', () => {
    const customName = enzyme.shallow(<CustomName name={'12345678901234567890123456789012'}/>);
    expect(customName).toMatchSnapshot();
});

it('33 long name is shortened after 30 characters', () => {
    const customName = enzyme.shallow(<CustomName name={'123456789012345678901234567890123'}/>);
    expect(customName).toMatchSnapshot();
});

it('Whitespace is trimmed before shortening', () => {
    const customName = enzyme.shallow(<CustomName name={'          1234567890123          '}/>);
    expect(customName).toMatchSnapshot();
});
