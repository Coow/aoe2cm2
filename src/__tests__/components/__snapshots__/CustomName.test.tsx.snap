// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`32 long name is not shortened 1`] = `
<span>
  12345678901234567890123456789012
</span>
`;

exports[`33 long name is shortened after 30 characters 1`] = `
<abbr
  title="123456789012345678901234567890123"
>
  123456789012345678901234567890…
</abbr>
`;

exports[`Long name is shortened 1`] = `
<abbr
  title="123456789012345678901234567890123456789"
>
  123456789012345678901234567890…
</abbr>
`;

exports[`Short name is not shortened 1`] = `
<span>
  Player Name
</span>
`;

exports[`Whitespace is trimmed before shortening 1`] = `
<span>
  1234567890123
</span>
`;
